src/main/java/io/confluent/idesidecar/restapi/proxy/FlinkDataPlaneAuthenticationProcessor.java

```diff
@@ -0,0 +1,46 @@
+package io.confluent.idesidecar.restapi.proxy;
+
+import static io.vertx.core.http.HttpHeaders.AUTHORIZATION;
+
+import io.confluent.idesidecar.restapi.connections.CCloudConnectionState;
+import io.confluent.idesidecar.restapi.exceptions.ProcessorFailedException;
+import io.confluent.idesidecar.restapi.processors.Processor;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import jakarta.enterprise.context.ApplicationScoped;
+
+/**
+ * Processor for handling data plane requests to services like Flink SQL and Catalog.
+ */
+@ApplicationScoped
+public class FlinkDataPlaneAuthenticationProcessor extends Processor<ProxyContext, Future<ProxyContext>> {
```

Comment:
This class does not do anything particularly Flink specific, it only sets the Confluent Cloud data plane token. We should rename this appropriately.
```suggestion
public class CCloudDataPlaneAuthenticationProcessor extends Processor<ProxyContext, Future<ProxyContext>> {
```